tests/at_end2end_test/test/notify_verb_test.dart

```diff
@@ -92,18 +95,22 @@ void main() {
       // Wait for a second
       await Future.delayed(Duration(seconds: 1));
 
-      // verify atsign2 receives all notifications ~immediately
-      for (int i = 1; i <= numNotifications; i++) {
-        print('/nWaiting for notification $i');
-        String subsequentNotification = await monitorSH.read(
-            log: true, timeoutMillis: 1000, throwTimeoutException: true);
-        print('Notification number $i: $subsequentNotification');
-        final notifJson = jsonDecode(
-            subsequentNotification.replaceFirst('notification: ', ''));
-
-        expect(sentNotificationIds, contains(notifJson['id']));
-        sentNotificationIds.remove(notifJson['id']);
+      List<String> toCheck = [...sentNotificationIds];
+      logger.info(
+          'Will wait for these notification IDs to be received: $toCheck');
+      // verify atsign2 receives all notifications
+      while (toCheck.isNotEmpty) {
+        String receivedNotif = await monitorSH.read(
+            log: true, timeoutMillis: 5000, throwTimeoutException: true);
+        final notifJson =
+            jsonDecode(receivedNotif.replaceFirst('notification: ', ''));
+        var notifId = notifJson['id'];
+        if (toCheck.contains(notifId)) {
+          logger.info('Received $notifId');
+          toCheck.remove(notifId);
+        }
       }
+      expect(toCheck, isEmpty);
```

Comment:
The rework above makes the test resilient to receiving "stats" notifications with id -1 - this caused a test run to fail earlier.